vortex-array/src/canonical.rs

```diff
@@ -523,18 +523,20 @@ impl IntoCanonical for ArrayData {
         if !self.is_canonical() && self.len() > 1 {
             log::trace!("Canonicalizing array with encoding {:?}", self.encoding());
         }
-        self.encoding().into_canonical(self)
+        self.encoding().clone().into_canonical(self)
```

Comment:
we can save on a bunch of these clones by changing `into_canonical`/`into_arrow` to take `self: Arc<Self>`